File: Persistencia/Ej2/Cliente.java
import java.util.ArrayList;
import java.util.List;

//3.
//a) Implementar el diagrama de clases.
public class Cliente {
    int id;
    String nombre;
    int telefono;

    public Cliente(int id, String nombre, int telefono) {
        this.id = id;
        this.nombre = nombre;
        this.telefono = telefono;
    }

    @Override
    public String toString() {
        return "Cliente{id=" + id + ", nombre='" + nombre + "', telefono=" + telefono + "}";
    }
}

class ArchivoCliente {
    String nomA;
    List<Cliente> clientes;

    public ArchivoCliente(String nomA) {
        this.nomA = nomA;
        this.clientes = new ArrayList<>();
    }

    public void guardaCliente(Cliente c) {
        clientes.add(c);
    }

    // b) Implementa buscarCliente(int c) a través del id.
    public Cliente buscarCliente(int id) {
        for (Cliente c : clientes) {
            if (c.id == id) {
                return c;
            }
        }
        return null;
    }

    // c) Implementa buscarCelularCliente(int c), que devuelva los datos del cliente
    // junto al número de celular.
    public Cliente buscarCelularCliente(int id) {
        return buscarCliente(id);
    }

    public void mostrarEjemplo() {
        Cliente c1 = new Cliente(1, "Emanuel", 7654321);
        Cliente c2 = new Cliente(2, "Luis", 9876543);
        guardaCliente(c1);
        guardaCliente(c2);

        System.out.println(buscarCliente(1));
        System.out.println(buscarCelularCliente(2));
    }
}

class Main {
    public static void main(String[] args) {
        ArchivoCliente archivo = new ArchivoCliente("clientes.txt");
        archivo.mostrarEjemplo();
    }
}

File: Persistencia/Ej2/Cliente.py
class Cliente:
    def __init__(self, id, nombre, telefono):
        self.id = id
        self.nombre = nombre
        self.telefono = telefono

    def __str__(self):
        return f"Cliente(id={self.id}, nombre={self.nombre}, telefono={self.telefono})"


class ArchivoCliente:
    def __init__(self, nomA):
        self.nomA = nomA
        self.clientes = []

    def guarda_cliente(self, cliente):
        self.clientes.append(cliente)

    def buscar_cliente(self, id):
        for cliente in self.clientes:
            if cliente.id == id:
                return cliente
        return None

    def buscar_celular_cliente(self, id):
        return self.buscar_cliente(id)

    def mostrar_ejemplo(self):
        c1 = Cliente(1, "Emanuel", 7654321)
        c2 = Cliente(2, "Luis", 9876543)
        self.guarda_cliente(c1)
        self.guarda_cliente(c2)

        print(self.buscar_cliente(1))
        print(self.buscar_celular_cliente(2))


if __name__ == "__main__":
    archivo = ArchivoCliente("clientes.txt")
    archivo.mostrar_ejemplo()